test/test_jsonplaceholder.py

```diff
@@ -204,3 +205,34 @@ def test_upload_file_accesses_the_right_endpoint_when_called(self):
                 headers={"Content-type": "application/json; charset=UTF-8"},
             )
             self.assertIs(api.upload_file.response, response)
+
+    def test_create_post_with_schema(self):
+        api = qrest.API(jsonplaceholderconfig)
+        api.create_post_with_schema.response = ContentResponse()
+
+        post = {'user': 'Alice', 'body': 'Something about bob'}
+
+        with mock.patch("requests.request", return_value=self.mock_response) as mock_request:
+            response = api.create_post_with_schema.get_response(post=post)
+
+            mock_request.assert_called_with(
+                method="POST",
+                auth=None,
+                verify=False,
+                url="https://jsonplaceholder.typicode.com/posts",
+                params={},
+                json=post,
+                files=[],
+                headers={"Content-type": "application/json; charset=UTF-8"},
+            )
+            self.assertIs(api.create_post_with_schema.response, response)
+
+    def test_bad_create_post_with_schema(self):
+        api = qrest.API(jsonplaceholderconfig)
+        api.create_post_with_schema.response = ContentResponse()
+
+        # parameter that does not obey the schema
+        post = {'user': 'Alice', 'message': 'Something about bob'}
+
+        with self.assertRaises(RestClientValidationError):
```

Comment:
Almost ;) Can you verify that the `RestClientValidationError` that you expected was actually raised (as this exception can be raised for numerous reasons).

Reply:
Will fix